mkosi.images/system/mkosi.conf.d/10-arch/mkosi.build.chroot

```diff
@@ -24,12 +24,31 @@ ln --symbolic . "pkg/$ID/src"
 # shellcheck source=/dev/null
 . /etc/makepkg.conf
 
+CFLAGS="$CFLAGS -O${OPTIMIZATION:-0}"
+if [[ "${OPTIMIZATION:-0}" == "0" ]]; then
+    CFLAGS="$CFLAGS -Wp,-U_FORTIFY_SOURCE"
```

Comment:
I think it would be great if it was possible to turn it on/off explicitly (to prevent it from interfering with Valgrind for example) and set it to, say, 3 explicitly too to avoid relying on distros defaults.

Reply:
@evverx See the commit where we now pick up $CFLAGS from `Environment=`. I don't want to add a separate option for each argument. Being able to override fortify source levels via $CFLAGS seems good enough to me.

Reply:
That commit wasn't there 20 minutes ago :-)

> I don't want to add a separate option for each argument

Makes sense. Though as far as I understand currently if I pass `CFLAGS='-O2 -D_FORTIFY_SOURCE=3'` without specifying `OPTIMIZATION` explicitly it gets overridden by `-O0 -Wp,-U_FORTIFY_SOURCE`. It's kind of counterintuitive I think.

Reply:
I'm not going to write a full blown compilation option parser in bash. We'll just have to document that optimization level should be set using `$OPTIMIZATION`

Reply:
Agreed. Though I think that it's still kind of unpredictable because by default it depends on whatever distros do as far as I understand. I think it would be easier to use it if was off by default and CFLAGS were used to turn it on explicitly.

Reply:
@DaanDeMeyer would it be possible to move FORTIFY_SOURCE to a separate PR? I think it would be better to make clang with ASan/UBSan work first and then figure out how it all should interact with things like https://github.com/google/sanitizers/issues/247

Reply:
Removed from this PR, will open another PR once this one is merged with the fortify changes